components/chart/index.tsx

```diff
@@ -55,8 +59,35 @@ function Chart({
     ZOOM_OUT: () => changeZoomLevel(10),
   };
 
+  const toggleDrawer = () => {
+    setDrawerOpen((prev) => !prev);
+  };
+  const drawerRef = useRef(null);
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (drawerRef.current && !drawerRef.current.contains(event.target) && drawerOpen) {
+        toggleDrawer();
+      }
+    };
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  });
+  const findModuleById = (id: number): string | undefined => {
+    const arrayOfNodes = Object.keys(currentVersion.nodes).map((key) => currentVersion.nodes[key]);
+    const module = arrayOfNodes.find((module) => module.id === id);
+    const indexOfOpeningParentheses = module.title.indexOf('(');
+    const cutName = module.title.substring(0, indexOfOpeningParentheses).trim();
+    setName(cutName);
+    setDescription(module.description);
+    return module ? module.description : undefined;
+  };
+
   const selectModule = useCallback(
     (id: number, groupSelect?: boolean) => {
+      setDrawerOpen(true);
+      findModuleById(id);
```

Comment:
I think findModuleById should return the node (module) then we can set the description and the name inside selectModule
a function should only do what it intended to do in our case it should find a module and return it only

By the way maybe it would be better to modify the place where we are calling select Module to pass also the module, but you can keep "findModuleById" function we might use it later

Reply:
done